src/v/storage/disk_log_impl.cc

```diff
@@ -1051,10 +1051,14 @@ disk_log_impl::maybe_apply_local_storage_overrides(gc_config cfg) const {
     /*
      * don't override with local retention settings--let partition data expand
      * up to standard retention settings.
+     * NOTE: both retention_local_strict and retention_local_strict_override
+     * need to be set to true to honor strict local retention. Otherwise, local
+     * retention is advisory.
      */
-    if (
```

Comment:
Thanks for the note above. I even had to write down the truth table to make sure it is correct. I did read the code a couple of times and still was confused...

Reply:
Nits:

`strict_retention` => `apply_local_retention`
`Skipped retention override for topic with remote write` => `Skipped local retention override for topic with remote write`

Reply:
I prefer `strict_retention` as a name. In general I prefer naming variables what they are rather than their intended outcome. It's easier for me to reason about why things are happening when expressed "I applied retention because of `strict_retention`" vs "I applied retention because of `apply_local_retention`". As a reader at least then I don't have to read the variable definition to figure out what policy is actually being enacted by `apply_local_retention` (I know the variable definition is right there, so it's not too important here)

Reply:
Will change to `strict_local_retention`